Vhar 7932 korvaa luoja urakka idlla

rckopteri opened the pull request.

rckopteri committed:
Korvaa luoja urakalla laatupoikkeamien hauista.

rckopteri committed:
Korvaa luoja urakalla laatupoikkeamien hauista.

rckopteri committed:
Korvaa luoja urakka-id:llä siltatarkastus hauissa.

rckopteri committed:
Korvaa luoja urakka-id:llä tarkastuksien api kyselyjen tietokantahauissa.

rckopteri committed:
Korvaa luoja urakka-id:llä toteumien api kyselyjen tietokantahauissa.

rckopteri committed:
Korvaa luoja urakka-id:llä turvallisuuspoikkeamien api kyselyjen tietokantahauissa.

rckopteri committed:
Korvaa luoja urakka-id:llä  api kyselyjen tietokantahauissa.

rckopteri committed:
Merge branch 'develop' into VHAR-7932-korvaa-luoja-urakka-idlla

rckopteri committed:
Varmista api testeissä, että eri käyttäjällä päivitys ja poisto toimii edelleen.

rckopteri committed:
Siivoa duplikaatit ja estä niiden tuleminen lisäämällä unique index

rckopteri committed:
Korjattu indexin poisto

jpluomajoki-solita reviewed (approved):
Koodi näyttää hyvältä, pitäisi korjata duplikaattien tuonti.
Pohdintana: Onko riskiä, että rajapintakäyttäjä lähettäisi vaikkapa vahingossa väärällä urakka-id:llä ja olemassa olevalla ulkoisella id:llä rivin, ja tällöin ylikirjoittaisi jonkun muun urakan tavaraa?

Jos on, tämä voitaisiin vielä mahdollisesti estää rakentamalla käyttäjä-tauluun jonkinlaisen relaation siitä, että kuka käyttäjä saa ylikirjoittaa kenen käyttäjän dataa.

rckopteri committed:
Merge branch 'develop' into VHAR-7932-korvaa-luoja-urakka-idlla

rckopteri committed:
Rajoita päivitettävät toteumat ja tarkastukset 2023-05 jälkeiseen aikaan

Vähentää tarkastuksissa rivi määrää melkoisesti

rckopteri committed:
Vielä rajoituksia siihen, miten indeksit luodaan.

Tietokannassa on paljon väärin luotuja tarkistuksia ja toteumia. Indeksien luonti ei onnistu, jos ei vähän sovella aikataulullisesti.

rckopteri committed:
Korjataan id -> toteuma

rckopteri committed:
Vielä 2071 -> 2017, muuten uniikkiutta ei koskaan tarkasteta
